Show protocol mismatch warning when request is https

oliverguenther opened the pull request:
![2020-02-05-15:51:02](https://user-images.githubusercontent.com/459462/73852364-541beb00-482f-11ea-8148-6b1df3a8630d.png)

oliverguenther committed:
Show protocol and hostname mismatch warning

Shows a warning when

- protocol is http or https and the setting is different
- host name does not match the request

ulferts commented on app/views/warning_bar/_protocol_mismatch.erb:
```diff
@@ -0,0 +1,12 @@
+    <strong>Protocol setting mismatch</strong>
+    <br/>
+    Your application is running with Protocol set to <code><%= Setting.protocol %></code>, but the request
+    is a <code><%= request.ssl? ? 'https' : 'http' %></code>  request.
```
typo: a -> an (I think). Although 'h' is not a vowel, it is pronounced as such, almost like an 'a', in the context of 'http/s'.

The whole partial is not i18ned

oliverguenther replied:
I actually wanted to skip I18n here because it's admin stuff anyway, but fine for me to translate it

ulferts reviewed (changes requested):
I like the proposed change. Should help users running into less bugs. We might want to extend this to also include the hostname as it also causes a whole series of bugs, but that can also be done in a different PR.

Please see my i18n remark for the necessary changes.

oliverguenther commented:
I addressed your remarks. @machisuji @crohr Do you foresee any situations where the hostname of a browser request would not match the host_name setting ? The internal host name of a server obviously might differ, but the externally reachable address should always be the same right?
